src/screens/DetailScreen.js

```diff
@@ -115,6 +124,19 @@ export const DetailScreen = ({ navigation, route }) => {
     refreshTime
   });
 
+  let Component = getComponent(query);
+
+  if (query === QUERY_TYPES.SUE.REQUESTS_WITH_SERVICE_REQUEST_ID) {
+    return (
+      <Component
+        data={sueDetailsData}
+        navigation={navigation}
+        fetchPolicy={fetchPolicy}
+        route={route}
+      />
+    );
+  }
```

Comment:
if you check the query for returning the component you do not need to make the query condition for getting the component before? this way you check for `query === QUERY_TYPES.SUE.REQUESTS_WITH_SERVICE_REQUEST_ID` twice.
if you want to return early for `query === QUERY_TYPES.SUE.REQUESTS_WITH_SERVICE_REQUEST_ID` you can return `SUEDetailScreen` directly or am i wrong? no other code relies on `SUEDetailScreen` being in `getComponent`?

Reply:
the whole data query is also not needed in this screen and can be moved to the SueDetailScreen. I will work on this in the detail branch. you do not need  to make changes on this here.

Reply:
ab713d2